### ISSUE TITLE
Branie pod uwagę ocen z ułamkami w liczeniu średniej 

### ISSUE THREAD

**Issue by playsenge** [wulkanowy/wulkanowy#2594]
** Czy Twoja prośba o funkcję jest związana z problemem? Proszę opisz.**
Moi nauczyciele na drugi semestr przepisują średnią z pierwszego. VULCAN po stronie nauczyciela liczy to normalnie, ale nie Wulkanowy.

** Opisz żądane rozwiązanie **
Dodać opcję do ustawień która parsuje oceny jako floaty/double czy coś tam i brać to pod uwagę w kalkulacji średniej (najlepiej żeby akceptował formaty x.yz oraz x,yz)

![Screenshot_2024-06-12-12-43-57-76_44d29c633002065c27ecfc3cbea145bd](https://github.com/wulkanowy/wulkanowy/assets/40773107/bd6f6316-4576-48a3-8e0f-de1ff0f5fb4a)

**Comment by mklkj** [comment]
Chyba nadal nie rozumiem. Możesz powtórzyć? Najlepiej na przykładzie, razem z tym co jest liczone i jaki miałby być wynik

**Comment by playsenge** [comment]
> Chyba nadal nie rozumiem. Możesz powtórzyć? Najlepiej na przykładzie, razem z tym co jest liczone i jaki miałby być wynik

Tak jak na screenie. Ocena 4,93 nie jest brana pod uwagę, średnia jest pokazywana jako 5.0, a nie (4*4.93+2*5+2*5)/(4+2+2)

**Comment by mklkj** [comment]
Pokaż jaka średnia pokazuje ci się na stronie www dziennika

**Comment by playsenge** [comment]
Mam wyłączoną w szkole ale u nauczyciela wyświetla normalnie 4,97 a nie 5,00

**Comment by tomaszk8266** [comment]
Na screenie jest 4,93

**Comment by playsenge** [comment]
ŚREDNIA OBLICZONA JEST 5,00 A POWINNA BYĆ 4,97

**Comment by mklkj** [comment]
Okej, chwilę mi to zajęło, ale już rozumiem. 4.93 to nie jest prawidłowa ocena (a przynajmniej nie według tego, co Wulkanowy uważa za prawidłową ocenę) i faktycznie trzeba by było robić dodatkowe sztuczki, żeby móc ją zacząć interpretować jako prawidłową ocenę. To pociągałoby za sobą całkiem sporo zmian, bo parsowaniem i interpretacją ocen zajmuje się obecnie sdk, a cały model przechowywania ocen wymaga ocen jako liczby całkowite. Aż mi się nie chce wierzyć, że po stronie nauczyciela jest to liczone prawidłowo

**Comment by playsenge** [comment]
Dokładnie. Nie mam zdjęć, ale po stronie nauczyciela jest tak liczone. A zresztą powinna to być tylko włączana/wyłączana funkcja w ustawieniach. To nie powinno być trudne aby to zaimplementować, tylko tryParse na każdej ocenie jeśli nie jest na liście tradycyjnych i jeśli działa to użyć

**Comment by tomaszk8266** [comment]
Sr bym traktował jako skrót od "średnia", tylko dlaczego ma ona wagę 4? Jakiś dziwny sposób liczenia średniej końcowej?

**Comment by playsenge** [comment]
U mnie w szkole robią tak, że przepisują z wagą 4 średnią z pierwszego semestru, a roczna jest brana tylko z drugiego.
